src/main/typescript/wcardinal/ui/d-table-body-row.ts

```diff
@@ -438,4 +438,17 @@ export class DTableBodyRow<
 		}
 		return 0;
 	}
+
+	resizeCell(index: number, newWidth: number): void {
+		const cells = this.children;
+		for (const cell of cells) {
+			if (this.isCell(cell) && cell.columnIndex === index) {
+				const ratio = newWidth / cell.width;
```

Comment:
The relation of `cell.width` and `cell.weight` is `cell.width = cell.weight / (cell.weight + the sum of the other cell weights)`. Not linear. Because of this, `cell.weight = cell.weight * ratio` isn't correct.